perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace_einfraTest.java

```diff
@@ -0,0 +1,159 @@
+		classInstance.checkAttributeSyntax(session, user, attributeToCheck);
+	}
+
+	@Test(expected = ConsistencyErrorException.class)
+	public void testCheckAttributeSemanticsDuplicateDetected() throws Exception {
+		System.out.println("testCheckAttributeSemanticsDuplicateDetected()");
+		when(session.getPerunBl().getUsersManagerBl().getUsersByAttribute(session, attributeToCheck, true)).thenReturn(new ArrayList<>(users));
+		attributeToCheck.setValue(attributeToCheck.getValue().toString().toUpperCase());
+		classInstance.checkAttributeSemantics(session, user, attributeToCheck);
+	}
+
+	@Test(expected = WrongReferenceAttributeValueException.class)
+	public void testCheckAttributeSemanticsWithNullAttribute() throws Exception {
+		System.out.println("testCheckAttributeSemanticsWithNullAttribute()");
+		attributeToCheck.setValue(null);
+		classInstance.checkAttributeSemantics(session, user, attributeToCheck);
+	}
+
+	@Test
+	public void testCheckAttributeSemanticsCorrectValue() throws Exception {
+		System.out.println("testCheckAttributeSemanticsCorrectValue()");
+		List<User> tmp = new ArrayList<>();
+		tmp.add(user);
+		when(session.getPerunBl().getUsersManagerBl().getUsersByAttribute(session, attributeToCheck, true)).thenReturn(tmp);
+		attributeToCheck.setValue(attributeToCheck.getValue().toString().toUpperCase());
+		classInstance.checkAttributeSemantics(session, user, attributeToCheck);
+		verify(session.getPerunBl().getUsersManagerBl()).getUsersByAttribute(session, attributeToCheck, true);
+		verify(session.getPerunBl().getUsersManagerBl()).checkReservedLogins(session,attributeToCheck.getFriendlyNameParameter(), attributeToCheck.valueAsString(), true);
+	}
+
+	@Test(expected = WrongReferenceAttributeValueException.class)
+	public void testCheckAttributeSemanticsAlreadyExistIgnoresCase() throws Exception {
+		System.out.println("testCheckAttributeSemanticsAlreadyExistIgnoresCase()");
+		List<User> tmp = new ArrayList<>();
+		tmp.add(user);
+		tmp.add(user2);
+		when(session.getPerunBl().getUsersManagerBl().getUsersByAttribute(session, attributeToCheck, true)).thenReturn(tmp);
+		attributeToCheck.setValue(attributeToCheck.getValue().toString().toUpperCase());
+		classInstance.checkAttributeSemantics(session, user, attributeToCheck);
+	}
+
+	@Test(expected = WrongReferenceAttributeValueException.class)
+	public void testAlreadyReservedLogin() throws Exception {
+		System.out.println("testAlreadyReservedLogin()");
+		UsersManagerBl usersManager = mock(UsersManagerBl.class);
+		when(session.getPerunBl().getUsersManagerBl()).thenReturn(usersManager);
+		List<User> tmp = new ArrayList<>();
+		tmp.add(user);
+		tmp.add(user2);
+		when(usersManager.getUsersByAttribute(session, attributeToCheck, true)).thenReturn(tmp);
+		doThrow(new AlreadyReservedLoginException(""))
+			.when(usersManager).checkReservedLogins(session, "einfra", attributeToCheck.valueAsString(), true);
+		attributeToCheck.setValue(attributeToCheck.getValue().toString().toUpperCase());
```

Comment:
In order to _doThrow()_ correctly work, you will have to move this line before _doThrow()_ or remove it since it is irrelevant.